test/test.js

```diff
@@ -458,6 +462,37 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+    pathname === '/browserWorkerTestReports') {
+
+    var body = '';
+
+    req.on('data', function (chunk) {
+      body += chunk;
+    });
+
+    req.on('end', function () {
+      var map = libCoverage.createCoverageMap(JSON.parse(body));
+      var summary = libCoverage.createCoverageSummary();
+      map.merge(summary);
```

Comment:
This line is not meaningful. The intended purpose of the code in this block is to:

1. Obtain a CoverageMap object for existing coverage data (i.e. from previously submitted data).
2. Add more data (i.e. from `body`).
3. Summarize the overall data and write it to a file.

Currently. your code is taking the existing body as a base (while you should be using the previous data, e.g. by reading the existing file, parsing it as a `CoverageMap` and merging it with the existing file).

Repeatedly reading and writing a file to generate coverage is inefficient, and that approach should only be used as a temporary method to experiment with coverage data collection. This optimization can be done at a later stage in the PR.